### PR TITLE
add div(gp) to MAC projection rhs

### PR DESCRIPTION
**Pull request by michaeljbrazell**
implemented Shreyas' idea to add divergence of gp so that mac projection solution phi looks more like pressure

### CODE REVIEW

File: amr-wind/equation_systems/icns/icns_ops.cpp [hunk 1]
```diff
@@ -97,7 +112,13 @@ void advection_mac_project(FieldRepo& repo, const FieldState fstate, const bool
         mac::get_projection_bc(
             amrex::Orientation::high, bctype, repo.mesh().Geom()));
 
-    macproj.project(options.rel_tol, options.abs_tol);
+    if(has_overset){
+        auto phif = repo.create_scratch_field(1, 1, amr_wind::FieldLoc::CELL);
+        amr_wind::field_ops::copy(*phif, repo.get_field("p"), 0, 0, 1, 1);
```
Comment (sayerhs) [inline amr-wind/equation_systems/icns/icns_ops.cpp]: We need to average from node to cell-center here?

File: amr-wind/equation_systems/icns/icns_ops.cpp [hunk 2]
```diff
@@ -78,15 +80,28 @@ void advection_mac_project(FieldRepo& repo, const FieldState fstate, const bool
     amrex::LPInfo lp_info;
     lp_info.setMaxCoarseningLevel(options.max_coarsen_level);
 
+    auto div = repo.create_scratch_field(1, 0);
+    if(has_overset){
+        // fixme what should we do about density?
+        amr_wind::fvm::divergence((*div),repo.get_field("gp"));
+        // fixme I think this should be positive dt, but maybe also try -dt?
```
Comment (sayerhs) [inline amr-wind/equation_systems/icns/icns_ops.cpp]: The sign seems correct, but I am unsure about `dt` here.

File: amr-wind/equation_systems/icns/icns_ops.cpp [hunk 3]
```diff
@@ -78,15 +80,28 @@ void advection_mac_project(FieldRepo& repo, const FieldState fstate, const bool
     amrex::LPInfo lp_info;
     lp_info.setMaxCoarseningLevel(options.max_coarsen_level);
 
+    auto div = repo.create_scratch_field(1, 0);
+    if(has_overset){
+        // fixme what should we do about density?
```
Comment (sayerhs) [inline amr-wind/equation_systems/icns/icns_ops.cpp]: The `div( grad(p) )` term should have the same coefficients as the `\beta` here: https://amrex-codes.github.io/amrex/docs_html/LinearSolvers.html#mac-projection

File: amr-wind/equation_systems/icns/icns_ops.cpp (hunk 1 above)
Comment (michaeljbrazell) [reply amr-wind/equation_systems/icns/icns_ops.cpp]: fixed it

**Review by sayerhs — APPROVED** [review]

**Pull request merged**